### PR TITLE
feat(admin-ui): Implement security measures for webhook 

### LINKED ISSUE #1704: feat(admin-ui): Implement security measures for webhook
-------------------

### Description

Currently the webhook feature doesn't have any security measures implemented.
The following should be done: 
-  Perform URL validation.
-  Ensure URL starts with "https://", disallow "file://" and other non-HTTPS schemes.
-  Block typical local IPs: 127.0.x, 192.168.x, 172.x.
-  Prohibit "localhost" and "http://"
-  Require a specific response header for POST requests, unique to the customer.

-------------------
image

### PR DESCRIPTION
**Pull request by syntrydy**
closed #1704

### CODE REVIEW

File: admin-ui/plugins/admin/components/Webhook/WebhookURLChecker.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const NOT_ALLOWED = ["http://", "ftp://", "file://", "telnet://", "smb://", "ssh://", "ldap://", "https://192.168", "https://127.0", "https://172", "https://localhost"]
+export const isValid = (url) => {
+    if (url === undefined || url === null) {
+        return false;
+    } else {
+        return isAllowed(url)
+    }
+}
+
+const isAllowed = (url) => {
```
Comment (jv18creator) [inline admin-ui/plugins/admin/components/Webhook/WebhookURLChecker.js:10]: I think if we add `https://local` as an input the `isValid` will return true even if the url is invalid. Could you check [this answer](https://stackoverflow.com/a/61691586/13718296) on stackoverflow. For validation they are using Yup as we do for other fields.

File: admin-ui/plugins/admin/components/Webhook/WebhookForm.js [hunk 2]
```diff
@@ -63,6 +64,14 @@ const WebhookForm = () => {
         )
       }
     }
+    const isCool = isValid(values.url)
```
Comment (jv18creator) [inline admin-ui/plugins/admin/components/Webhook/WebhookForm.js:67]: Please remove unused variable.

**Review by jv18creator — CHANGES_REQUESTED** [review]
Please check the inline comments.

File: admin-ui/plugins/admin/components/Webhook/WebhookURLChecker.js (hunk 1 above)
Comment (syntrydy) [reply admin-ui/plugins/admin/components/Webhook/WebhookURLChecker.js:10]: @jv18creator the validation we need on the Ui is just to block unwanted urls, we don't want to check if the url is resolvable or no. 

That check will be done on the backend when the webhook is enable and about to be schedule.

File: admin-ui/plugins/admin/components/Webhook/WebhookForm.js (hunk 2 above)
Comment (syntrydy) [reply admin-ui/plugins/admin/components/Webhook/WebhookForm.js:67]: Done

File: admin-ui/plugins/admin/components/Webhook/WebhookURLChecker.js (hunk 1 above)
Comment (syntrydy) [reply admin-ui/plugins/admin/components/Webhook/WebhookURLChecker.js:19]: But i have added the Regex. Regading Yup in this case i don't think we need it as we have complexe validation, regex and blacklist).

**Review by jv18creator — APPROVED** [review]

**Pull request merged**